Makefile

```diff
@@ -102,7 +102,7 @@ ci-clean:
 # override COMMIT.
 .PHONY: deploy
 deploy:
-	go run -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro deploy dev-config.yaml ${LOCATION}
+	go run -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(COMMIT)" ./cmd/aro deploy dev-config.yaml ${LOCATION}
```

Comment:
I think we should not do that because in several parts of our prod pipelines we reference `make deploy`, so we're using this in prod with the TAG and we would likely break things.

Reply:
I added a condition on FULL_RP_DEV which will be set whenever we wish to always use COMMIT. It would be useful for the full RP automation #3764